modules/apps/portal-configuration/portal-configuration-module-configuration-api/src/main/java/com/liferay/portal/configuration/module/configuration/ConfigurationProvider.java

```diff
@@ -59,6 +59,10 @@ public <T> void saveCompanyConfiguration(
 			Dictionary<String, Object> properties)
 		throws ConfigurationException;
 
+	public <T> void saveCompanyConfiguration(
+			long companyId, String pid, Dictionary<String, Object> properties)
+		throws ConfigurationException;
+
```

Comment:
Why are you adding this method? It looks like this is only used in one of your tests, and the method signature is inconsistent with all other in the same class; why don't you create a inner class in the test with the expected annotations and use one of the existing methods?

Reply:
Hi, this is complex
On the test you can not use the IFrameConfiguration class.
It does not have visualization and you can not export the class on the portal-security-iframe-sanitizer module because the SF fails because the name does not end on the allowed ones.

So if you can not import the class you can only use the PID on the test module to check the configuration.

So I realized that we have this method on this class
```

public <T> void saveGroupConfiguration(
			long groupId, String pid, Dictionary<String, Object> properties)
		throws ConfigurationException;
```

so taking this into account and that only is used once I decided to create this method
```

public <T> void saveCompanyConfiguration(
			long companyId, String pid, Dictionary<String, Object> properties)
		throws ConfigurationException;
```

If this is a No, we should recheck all the solution because if not we can not test it because the ConfigurationTemporarySwapper or ConfigurationCompanyTemporarySwapper does not trigger the update of the helper/managedService and you can not test it.


Any thought?

Reply:
But you can create a fake interface, internal to your test class, with with expected factoryId,method names and annotations. That's what I was suggesting above.

Reply:
talked about on slack.